gluten-core/src/main/scala/org/apache/spark/shuffle/sort/ColumnarShuffleManager.scala

```diff
@@ -143,44 +107,29 @@ class ColumnarShuffleManager(conf: SparkConf) extends ShuffleManager with Loggin
         shouldBatchFetch = shouldBatchFetch
       )
     } else {
-      new BlockStoreShuffleReader(
-        handle.asInstanceOf[BaseShuffleHandle[K, _, C]],
-        blocksByAddress,
+      sortShuffleManager.getReader(
+        handle,
+        startMapIndex,
+        endMapIndex,
+        startPartition,
+        endPartition,
         context,
-        metrics,
-        shouldBatchFetch = shouldBatchFetch
-      )
+        metrics)
     }
   }
 
   /** Remove a shuffle's metadata from the ShuffleManager. */
   override def unregisterShuffle(shuffleId: Int): Boolean = {
-    Option(taskIdMapsForShuffle.remove(shuffleId)).foreach {
-      mapTaskIds =>
-        mapTaskIds.iterator.foreach {
-          mapId => shuffleBlockResolver.removeDataByMap(shuffleId, mapId)
-        }
-    }
-    true
+    sortShuffleManager.unregisterShuffle(shuffleId)
```

Comment:
Does it same with before ? The `taskIdMapsForShuffle` is maintained by `ColumnarShuffleManager` so `sortShuffleManager. unregisterShuffle` will do nothing since the `taskIdMapsForShuffle` is empty.

Reply:
I missed this thanks, updated shuffleBlockResolver related flow.